src/values/UnDeserializableValue.js

```diff
@@ -94,21 +94,26 @@ var SELF = dv.UnDeserializableValue = util.inherit(
 	},
 
 	/**
-	 * Returns the error object stating why some unserializer was not able to unserialize the
+	 * Returns the error object stating why some deserializer was not able to deserialize the
 	 * structure.
 	 *
-	 * @return {Error}
+	 * @return {string}
 	 */
 	getReason: function() {
-		return this._unserializeError;
+		return this._deserializeError;
 	},
 
 	/**
 	 * @inheritdoc
 	 */
 	equals: function( other ) {
-		// TODO: Do deep equal of the structures and reasons instead.
-		return this === other;
+		if ( !( other instanceof SELF ) ) {
+			return false;
+		}
+
+		return this.getTargetType() === other.getTargetType()
+			&& JSON.stringify( this.getStructure() ) === JSON.stringify( other.getStructure() )
+			&& this.getReason() === other.getReason();
```

Comment:
Suggestion: `return JSON.stringify( this.toJSON() ) === JSON.stringify( other.toJSON() );`.